File: src/plugin/combat/magic/ancient/teleport/GhorrockTeleportSpell.java
package plugin.combat.magic.ancient.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/27/2017
 */
public class GhorrockTeleportSpell extends TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 96;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(LAW_RUNE, 2, WATER_RUNE, 8);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2977, 3873, 0);
	}
	
	@Override
	public int spellId() {
		return 47;
	}
	
	@Override
	public double exp() {
		return 106;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.ANCIENTS;
	}
}

File: src/main/kotlin/plugin/interaction/combat/magic/ancient/teleport/GhorrockTeleportSpell.kt
package plugin.interaction.combat.magic.ancient.teleport

import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin
import org.redrune.game.global.WorldTile
import org.redrune.utility.constants.MagicConstants
import org.redrune.utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 7/27/2017
 */
class GhorrockTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 96
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.LAW_RUNE, 2, MagicConstants.WATER_RUNE, 8)
    }

    override fun destination(): WorldTile {
        return WorldTile(2977, 3873, 0)
    }

    override fun spellId(): Int {
        return 47
    }

    override fun exp(): Double {
        return 106.0
    }

    override fun book(): MagicBook {
        return MagicBook.ANCIENTS
    }
}